src/errors.rs

```diff
@@ -0,0 +1,48 @@
+//! Module for all GeoJSON-related errors
+use serde_json::value::Value;
+use thiserror::Error;
+use crate::geometry::Value as GValue;
+
+/// Errors which can occur when encoding, decoding, and converting GeoJSON
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Encountered non-array value for a 'bbox' object: `{0}`.")]
+    BboxExpectedArray(Value),
+    #[error("Encountered non-numeric value within 'bbox' array.")]
+    BboxExpectedNumericValues(Value),
+    #[error("Encountered a non-object type for GeoJSON: `{0}`.")]
+    GeoJsonExpectedObject(Value),
+    /// This was previously `GeoJsonUnknownType`, but has been split for clarity
+    #[error("Expected a Feature, FeatureCollection, or Geometry, but got an empty type.")]
+    EmptyType,
+    /// This was previously `GeoJsonUnknownType`, but has been split for clarity
+    #[error("Expected a Feature mapping, but got a `{0}`.")]
+    NotAFeature(String),
+    #[error("Encountered a mismatch when converting to a Geo type: `{0}`.")]
```

Comment:
very minor, but what do you think about removing periods for consistency with the latter error strings?

```suggestion
    #[error("Encountered non-array value for a 'bbox' object: `{0}`")]
    BboxExpectedArray(Value),
    #[error("Encountered non-numeric value within 'bbox' array")]
    BboxExpectedNumericValues(Value),
    #[error("Encountered a non-object type for GeoJSON: `{0}`")]
    GeoJsonExpectedObject(Value),
    /// This was previously `GeoJsonUnknownType`, but has been split for clarity
    #[error("Expected a Feature, FeatureCollection, or Geometry, but got an empty type")]
    EmptyType,
    /// This was previously `GeoJsonUnknownType`, but has been split for clarity
    #[error("Expected a Feature mapping, but got a `{0}`")]
    NotAFeature(String),
    #[error("Encountered a mismatch when converting to a Geo type: `{0}`")]
```